src/modules/auth/auth.controller.ts

```diff
@@ -1,27 +1,110 @@
-import { Body, Controller, Post } from '@nestjs/common';
+import {
+  Body,
+  Controller,
+  Get,
+  Post,
+  Req,
+  Res,
+  UnauthorizedException,
+} from '@nestjs/common';
 import { RegisterDto } from './dto/request/register.dto';
 import { AuthService } from './auth.service';
 import { LoginDto } from './dto/request/login.dto';
+import { Request, Response } from 'express';
+import { ConfigService } from '@nestjs/config';
+import { isProd } from 'src/shared/helpers';
+import ms from 'ms';
 
 @Controller('auth')
 export class AuthController {
-  constructor(private readonly authService: AuthService) {}
+  constructor(
+    private readonly authService: AuthService,
+    private readonly configService: ConfigService,
+  ) {}
 
   @Post('register')
-  async register(@Body() registerDto: RegisterDto) {
-    const accessToken = await this.authService.register(registerDto);
+  async register(
+    @Body() registerDto: RegisterDto,
+    @Res({ passthrough: true }) res: Response,
+  ) {
+    const tokens = await this.authService.register(registerDto);
 
-    return {
-      data: accessToken,
-    };
+    res.cookie('accessToken', tokens.accessToken, {
+      httpOnly: true,
+      secure: isProd(this.configService.get('NODE_ENV')),
+      maxAge: ms(
+        this.configService.get<string>('AUTH_ACCESS_TOKEN_EXPIRES_IN'),
+      ),
+    });
+    res.cookie('refreshToken', tokens.refreshToken, {
+      httpOnly: true,
+      secure: isProd(this.configService.get('NODE_ENV')),
+      maxAge: ms(
+        this.configService.get<string>('AUTH_REFRESH_TOKEN_EXPIRES_IN'),
+      ),
+    });
   }
 
   @Post('login')
-  async login(@Body() loginDto: LoginDto) {
-    const accessToken = await this.authService.login(loginDto);
+  async login(
+    @Body() loginDto: LoginDto,
+    @Res({ passthrough: true }) res: Response,
+  ) {
+    const tokens = await this.authService.login(loginDto);
 
-    return {
-      data: accessToken,
-    };
+    res.cookie('accessToken', tokens.accessToken, {
+      httpOnly: true,
+      secure: isProd(this.configService.get('NODE_ENV')),
+      maxAge: ms(
+        this.configService.get<string>('AUTH_ACCESS_TOKEN_EXPIRES_IN'),
+      ),
+    });
+    res.cookie('refreshToken', tokens.refreshToken, {
+      httpOnly: true,
+      secure: isProd(this.configService.get('NODE_ENV')),
+      maxAge: ms(
+        this.configService.get<string>('AUTH_REFRESH_TOKEN_EXPIRES_IN'),
+      ),
+    });
+  }
+
+  @Get('refresh')
+  async refresh(
+    @Req() req: Request,
+    @Res({ passthrough: true }) res: Response,
+  ) {
+    const refreshToken = req.cookies['refreshToken'];
```

Comment:
I think you should create guard and strategy with passport for refresh token logic.

Reply:
fixed